ladybug/sunpath.py

```diff
@@ -80,34 +88,63 @@ def from_location(cls, location, north_angle=0, daylight_saving_period=None):
 
     @property
     def latitude(self):
-        """Get/set latitude in degrees."""
+        """Get or set a number between -90 and 90 for the latitude in degrees."""
         return math.degrees(self._latitude)
 
     @latitude.setter
     def latitude(self, value):
-        """Set latitude value."""
         self._latitude = math.radians(float(value))
         assert -self.PI / 2 <= self._latitude <= self.PI / 2, \
-            "latitude value should be between -90..90."
+            'latitude value should be between -90 and 90. Got {}.'.format(value)
+        if self._latitude == self.PI / 2:  # prevent math domain errors
+            self._latitude = self._latitude - 1e-9
+        if self._latitude == -self.PI / 2:  # prevent math domain errors
+            self._latitude = self._latitude + 1e-9
 
     @property
     def longitude(self):
-        """Get longitude in degrees."""
+        """Get or set a number between -180 and 180 for the longitude in degrees.
+
+        Note that you will also likely want to update the time zone of the
+        Sunpath if this value is set to somethign far from its original value.
+        """
         return math.degrees(self._longitude)
 
     @longitude.setter
     def longitude(self, value):
-        """Set longitude value in degrees."""
         self._longitude = math.radians(float(value))
+        assert -self.PI <= self._longitude <= self.PI, \
+            'longitude value should be between -180 and 180. Got {}.'.format(value)
+
+    @property
+    def time_zone(self):
+        """Get or set the time zone as a number between -12 and + 12.
+        
+        Setting this property to None will automatically set the Sunpath to use
+        solar time for the Sunpath's longitude.
+        """
+        return self._time_zone
 
-        # update time_zone
-        if abs((value / 15.0) - self.time_zone) > 1:
-            # if time_zone doesn't match the longitude update the time_zone
-            self.time_zone = value / 15.0
+    @time_zone.setter
+    def time_zone(self, tz):
+        self._time_zone = round(self.longitude / 15) if tz is None else float(tz)
+        assert -12 <= self._time_zone <= 12, \
```

Comment:
If we agree to change maximum to 14 then we have to update it here too. We should also update it in all the docstrings. 

```suggestion
        assert -12 <= self._time_zone <= 14, \
```

Reply:
Done!